cmd/esc/cli/env_run.go

```diff
@@ -163,6 +163,39 @@ func newEnvRunCmd(envcmd *envCommand) *cobra.Command {
 				}
 			}
 
+			writeTempFile := func(content string) (string, error) {
+				filename, f, err := envcmd.esc.fs.CreateTemp("", "esc-*")
+				if err != nil {
+					return "", err
+				}
+				defer contract.IgnoreClose(f)
+
+				if _, err = f.Write([]byte(content)); err != nil {
+					contract.IgnoreClose(f)
+					rmErr := envcmd.esc.fs.Remove(filename)
+					contract.IgnoreError(rmErr)
+					return "", err
+				}
+
+				return filename, nil
+			}
+			if files, ok := env.Properties["files"].Value.(map[string]esc.Value); ok {
+				for k, v := range files {
+					if strValue, ok := v.Value.(string); ok {
+						file, err := writeTempFile(strValue)
+						if err != nil {
+							return err
+						}
+						defer func() {
+							rmErr := envcmd.esc.fs.Remove(file)
+							contract.IgnoreError(rmErr)
+						}()
+
+						environ = append(environ, fmt.Sprintf("%v=%v", k, file))
+					}
+				}
+			}
```

Comment:
I think that we should also write these files in `open` for `--format=shell` and `--format=dotenv`.

Reply:
(LMK if you want to have a go at that. if not we can take this as-is and add support for `open` in a subsequent PR)

Reply:
I would prefer that we tackle it in a follow-up, to keep this PR as focused as possible.